tools/mason/MasonUtils.chpl

```diff
@@ -26,6 +26,8 @@ use TOML;
 use Path;
 use MasonEnv;
 
+config const useBash=false;
```

Comment:
I think configs like this should be handled through mason's arg-parsing mechanism, or just rely on an env var. It might be safe to rely on `$SHELL`.

Reply:
I don't think you can just rely on $SHELL here -- you ideally don't want to run this through the user shell here. 

One option to automatically do this is to use the solution here:
https://askubuntu.com/questions/97522/how-can-my-script-determine-whether-its-being-run-by-bash-or-dash 

Or just look at what /bin/sh points to and switch to bash if it looks like dash. That might be the easiest.

All of this is temporary, and should get fixed as soon as the Spack issue gets closed.